### ISSUE TITLE
AWS Gateway Load Balancer

### ISSUE THREAD

**Issue by aidansteele** [1debit/alternat#30]
AWS have an offering called the Gateway Load Balancer. Pricing is based on the [GWLB itself](https://aws.amazon.com/elasticloadbalancing/pricing/) and [GWLB endpoints](https://aws.amazon.com/privatelink/pricing/) (GWLBe). Both have per-hour and per-GB costs.

A single GWLB can be deployed to multiple AZs (just like an LB) and GWLBes are AZ-specific. So for a single AZ it would cost $0.0225/hr. Per-GB it would be an extra $0.0075 - this is assuming costs are dominated by bandwidth and not the other GLCU units (active connections and new connections).

As per [tweet](https://twitter.com/__steele/status/1590541408817274881) the main benefits would be high availability within each AZ (via multiple instances behind a GWLBe) and the ability for connection draining. 

There is even a [recent blog post](https://aws.amazon.com/blogs/networking-and-content-delivery/how-to-integrate-linux-instances-with-aws-gateway-load-balancer/) demonstrating exactly how this could be achieved. It is the "2-arm" mode that would most closely mirror how alterNAT works today.

**Comment by eddycek** [comment]
Hi @aidansteele I was looking at a blog post from AWS and it looks interesting, but not production ready. By any chance is there already some demo with alternat?

**Comment by bwhaley** [comment]
@eddycek There is a demo in [this video](https://www.youtube.com/watch?v=KtODOJPSz8A&t=1960s). Let me know if that's useful for you.

**Comment by eddycek** [comment]
@bwhaley Sorry, I meant the alterNAT demo with GWLB.

**Comment by nitrocode** [comment]
Would the gateway load balancer avoid needing to manage ips in the route table via the lambda? If so, then it would eliminate the issue of closed tcp connections during a scaling or max refresh event.

**Comment by eddycek** [comment]
> Would the gateway load balancer avoid needing to manage ips in the route table via the lambda? If so, then it would eliminate the issue of closed tcp connections during a scaling or max refresh event.

Yes it's right. The question is, how much would the price of operating the alternat with LB vs manage NAT increase.
